### PR TITLE
Follow up fix for ch7372

### PR DESCRIPTION
**Pull request by ihnorton**

### CODE REVIEW

File: tiledb/dataframe_.py [hunk 1]
```diff
@@ -569,6 +573,20 @@ def from_pandas(uri, dataframe, **kwargs):
...
+    elif coords_filters is None:
+        # empty coords filters support
+        coords_filters = tiledb.FilterList()
+
+    if offsets_filters is True:
+        offsets_filters = None
+    elif offsets_filters is None:
+        # empty offsets filters support
+        offsets_filters = tiledb.FilterList()
+
```
Comment (gsakkis) [inline tiledb/dataframe_.py:588]: 1. Can `coords_filters` and `offsets_filters` be a list? If yes shouldn't they be wrapped in a ` tiledb.FilterList`?
2. Unlike `attr_filters` and `dim_filters` no ValueError is raised here for invalid filters.

File: tiledb/dataframe_.py [hunk 2]
```diff
@@ -383,7 +384,9 @@ def create_dims(
             dim_filters = tiledb.FilterList(filters)
         elif isinstance(filters, tiledb.FilterList):
             dim_filters = filters
-        elif filters or filters is None:
+        elif filters is None:
+            dim_filters = tiledb.FilterList()
+        elif filters:
             dim_filters = None
         else:
```
Comment (gsakkis) [inline tiledb/dataframe_.py:391]: Is this deliberately inconsistent with the `attr_filters` handling? 
```
if attr_filters is True:
    # default case
    filters = tiledb.FilterList([tiledb.ZstdFilter(-1, ctx=ctx)], ctx=ctx) 
...
elif attr_filters is None:
    filters = None
```

**Review by gsakkis — CHANGES_REQUESTED** [review]

File: tiledb/dataframe_.py (hunk 1 above)
Comment (ihnorton) [reply tiledb/dataframe_.py:577]: Updated, thanks.

File: tiledb/dataframe_.py (hunk 2 above)
Comment (ihnorton) [reply tiledb/dataframe_.py]: Updated

File: tiledb/dataframe_.py [hunk 3]
```diff
@@ -176,25 +177,34 @@ def _get_column_infos(df, column_types, varlen_types):
...
+    elif isinstance(filters, tiledb.libtiledb.Filter):
+        return tiledb.FilterList([filters])
+    else:
+        raise ValueError("Unknown FilterList type!")
+
+
+def _get_attr_dim_filters(name, filters):
+    if isinstance(filters, dict):
+        # support passing a dict of filters per-attribute
+        return filters.get(name, None)
```
Comment (gsakkis) [inline tiledb/dataframe_.py:198]: I'd suggest `_get_schema_filters(filters.get(name, True))` to support all cases `_get_schema_filters` handles

File: tiledb/dataframe_.py [hunk 4]
```diff
@@ -524,7 +525,8 @@ def from_pandas(uri, dataframe, **kwargs):
     mode = tiledb_args.get("mode", "ingest")
     attr_filters = tiledb_args.get("attr_filters", True)
     dim_filters = tiledb_args.get("dim_filters", True)
-    coords_filters = tiledb_args.get("coords_filters", None)
+    coords_filters = tiledb_args.get("coords_filters", True)
+    offsets_filters = tiledb_args.get("offsets_filters", True)
```
Comment (gsakkis) [inline tiledb/dataframe_.py:529]: Not part of this PR but as a future TODO refactoring: remove the duplication of defaults between these `.get()` calls and `TILEDB_KWARG_DEFAULTS`

**Review by gsakkis — APPROVED** [review]

File: tiledb/dataframe_.py (hunk 3 above)
Comment (ihnorton) [reply tiledb/dataframe_.py]: ✔️

**Pull request merged**